rich/_log_render.py

```diff
@@ -15,7 +15,7 @@ def __init__(
         show_time: bool = True,
         show_level: bool = False,
         show_path: bool = True,
-        time_format: str = "[%x %X]",
+        time_format: Union[str, Callable[[datetime], str]] = "[%x %X]",
```

Comment:
The return type could be `Text` here, for flexibility.

Also, would you mind creating a type at module level, something like this:

```python
TimeFormatterCallable = Callable[[datetime], Text]
```

and then this param may be type as `Union[str, TimeFormatterCallable]`. Doesn't change anything, but I prefer it for readability sake

Reply:
I like the use of the type. If I pass in a text I get an exception:
```
  File "/Users/finn/src/finn/rich/rich/logging.py", line 141, in emit
    log_renderable = self.render(
  File "/Users/finn/src/finn/rich/rich/logging.py", line 187, in render
    log_renderable = self._log_render(
  File "/Users/finn/src/finn/rich/rich/_log_render.py", line 64, in __call__
    row.append(Text(log_time_display))
  File "/Users/finn/src/finn/rich/rich/text.py", line 139, in __init__
    self._text = [strip_control_codes(text)]
  File "/Users/finn/src/finn/rich/rich/control.py", line 47, in strip_control_codes
    return text.translate(_translate_table)
AttributeError: 'Text' object has no attribute 'translate'
```